### PR TITLE
feat: add typescript example from new expo init

### LINKED ISSUE #81: TypeScript example
- migrate and delete https://github.com/expo/typescript-starter-example

### PR DESCRIPTION
**Pull request by byCedric**
Fixes #81

I cheated a bit by not using the existing, but slightly outdated, [example repo](https://github.com/expo/typescript-starter-example). Created a new template with `expo init --template expo-template-blank-typescript` and added a readme, based on the expo docs.

### CODE REVIEW

File: with-typescript/.expo-shared/assets.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "f9155ac790fd02fadcdeca367b02581c04a353aa6d5aa84409a59f6804c87acd": true,
```
Comment (EvanBacon) [inline with-typescript/.expo-shared/assets.json]: remove `with-typescript/.expo-shared/assets.json`

File: with-typescript/app.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "expo": {
+    "name": "with-typescript",
+    "slug": "with-typescript",
```
Comment (EvanBacon) [inline with-typescript/app.json]: delete most of these fields, a good example is `with-preact/app.json`

File: with-typescript/app.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "privacy": "public",
+    "sdkVersion": "36.0.0",
+    "platforms": [
+      "ios",
+      "android",
+      "web"
+    ],
+    "version": "1.0.0",
+    "orientation": "portrait",
+    "icon": "./assets/icon.png",
```
Comment (EvanBacon) [inline with-typescript/app.json]: use remote assets and delete the local assets

File: with-typescript/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "main": "node_modules/expo/AppEntry.js",
```
Comment (EvanBacon) [inline with-typescript/package.json]:
```suggestion
```
remove this

File: with-typescript/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  "scripts": {
+    "start": "expo start",
+    "android": "expo start --android",
+    "ios": "expo start --ios",
+    "web": "expo start --web",
+    "eject": "expo eject"
```
Comment (EvanBacon) [inline with-typescript/package.json]: delete all scripts

File: with-typescript/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    "react-native-web": "~0.11.7"
+  },
+  "devDependencies": {
+    "@types/react": "~16.9.0",
+    "@types/react-native": "~0.60.23",
+    "@babel/core": "^7.0.0",
+    "babel-preset-expo": "~8.0.0",
+    "typescript": "~3.6.3"
+  },
+  "private": true
```
Comment (EvanBacon) [inline with-typescript/package.json]:
```suggestion
```

**Comment by byCedric** [comment]
Will do tonight!

**Comment by EvanBacon** [comment]
Right now I'm planning on stripping lock files, this may turn out to be a mistake in the future.

**Pull request merged**